### PR TITLE
doc: zigbee: clarify zigbee def handler docs

### PR DESCRIPTION
**Pull request by greg-fer**
Improved documentation of application utilities.
Moved sections around, added additional info.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
TODO: Add links to signals in ZBOSS.

File: include/zigbee/zigbee_app_utils.rst [hunk 1]
```diff
@@ -58,21 +58,16 @@ In general, using the default signal handler is worth considering because of the
...
-For example, when the application wants to use the procedure of the initiator of finding & binding or use the production configuration feature.
-
-Extending zboss_signal_handler
-++++++++++++++++++++++++++++++
+For example, when the application wants to use the procedure of the initiator of Finding & Binding or use the production configuration feature.
 
-If you want to extend ``zboss_signal_handler`` to cover additional functionalities, write signal handler implementation for the required ZBOSS signal.
-For example for a device that initiates a F&B procedure, extend ``zboss_signal_handler`` with a case for ``ZB_BDB_SIGNAL_FINDING_AND_BINDING_INITIATOR_FINISHED``:
+If you want to extend `zboss_signal_handler()`_ to cover additional functionalities, write signal handler implementation for the required ZBOSS signal.
+For example, for a device that initiates the F&B procedure, extend ``zboss_signal_handler`` with a switch case statement for ``ZB_BDB_SIGNAL_FINDING_AND_BINDING_INITIATOR_FINISHED``:
```
Comment (sebastiandraus) [inline include/zigbee/zigbee_app_utils.rst:70]: Do we want to change \`\`zboss_signal_handler\`\` to \`zboss_signal_handler()\`_ here and in other occurencies?

File: include/zigbee/zigbee_app_utils.rst [hunk 2]
```diff
@@ -130,109 +125,19 @@ For example for a device that initiates a F&B procedure, extend ``zboss_signal_h
...
+Complete zigbee_default_signal_handler implementation
+=====================================================
 
-        if (bufid) {
-            zb_buf_free(bufid);
-        }
-    }
+In its complete implementation, the ``zboss_signal_handler`` allows the application to control a broader set of basic functionalities, including joining, commissioning, and network formation.
 
+The following sections describe the logic implemented inside s:c:func:`zigbee_default_signal_handler`.
```
Comment (sebastiandraus) [inline include/zigbee/zigbee_app_utils.rst:135]: Typo?
```suggestion
The following sections describe the logic implemented inside :c:func:`zigbee_default_signal_handler`.
```

File: doc/nrf/links.txt [hunk 3]
```diff
@@ -143,6 +143,15 @@
...
+.. _`ZB_BDB_SIGNAL_FINDING_AND_BINDING_INITIATOR_FINISHED`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.6.0.0/group__zb__comm__signals.html#ga18f5e7ddfefb7d01ef98f696a9694d79
+.. _`ZB_ZDO_SIGNAL_PRODUCTION_CONFIG_READY`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.6.0.0/group__zgp__sink.html#ga3eb4631c12c345991be2461566c150e9
+.. _`ZB_ZDO_SIGNAL_SKIP_STARTUP`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.6.0.0/group__zb__comm__signals.html#gab2b9e7a13fd471d7bb642655d825a04e
+.. _`ZB_BDB_SIGNAL_DEVICE_FIRST_START`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.6.0.0/group__zb__comm__signals.html#gab55565980ef0580712f8dc160b01ea06
+.. _`ZB_BDB_SIGNAL_DEVICE_REBOOT`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.6.0.0/group__zb__comm__signals.html#ga3e9d86f42d2c6c27240f423a40fb152b
+.. _`ZB_BDB_SIGNAL_FORMATION`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.6.0.0/group__zb__comm__signals.html#ga342d4176c8932ff8ab0d8d6f997fa603
+.. _`ZB_BDB_SIGNAL_STEERING`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.6.0.0/group__zb__comm__signals.html#ga25a801841c95d6daddb9c60b39542b13v
```
Comment (sebastiandraus) [inline doc/nrf/links.txt:152]: Incorrect link
```suggestion
.. _`ZB_BDB_SIGNAL_STEERING`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.6.0.0/group__zb__comm__signals.html#ga25a801841c95d6daddb9c60b39542b13
```

**Review by sebastiandraus — CHANGES_REQUESTED** [review]

**Review by rlubos — APPROVED** [review]

**Review by sebastiandraus — APPROVED** [review]
LGTM

**Review by tomchy — APPROVED** [review]

**Pull request merged**